src/conn_sock.c

```diff
@@ -165,40 +285,63 @@ static gboolean read_conn_sock(struct conn_sock_s *sock)
 		return G_SOURCE_REMOVE;
 	}
 
-	num_read = read(sock->fd, sock->buf, CONN_SOCK_BUF_SIZE);
+	if (SOCK_IS_STREAM(sock->sock_type)) {
+		num_read = read(sock->fd, sock->buf, CONN_SOCK_BUF_SIZE);
+	} else {
+		num_read = recvfrom(sock->fd, sock->buf, CONN_SOCK_BUF_SIZE - 1, 0, NULL, NULL);
+	}
+
 	if (num_read < 0)
 		return G_SOURCE_CONTINUE;
 
 	if (num_read == 0)
-		return terminate_conn_sock(sock);
+		return terminate_remote_sock(sock);
 
 	/* num_read > 0 */
 	sock->remaining = num_read;
 	sock->off = 0;
 
-	sock_try_write_to_mainfd_stdin(sock);
+	if (SOCK_IS_NOTIFY(sock->sock_type)) {
+		/* Do what OCI runtime does - only pass READY=1 */
+		sock->buf[num_read] = '\0';
+		if (strstr(sock->buf, "READY=1")) {
+			strncpy(sock->buf, "READY=1", 8);
+			sock->remaining = 7;
+		} else {
+			sock->remaining = 0;
+		}
+	}
+
+	if (sock->remaining)
+		sock_try_write_to_local_sock(sock);
 
-	/* Not everything was written to stdin, let's wait for the fd to be ready.  */
+	/* Not everything was written, let's wait for the fd to be ready.  */
 	if (sock->remaining)
-		schedule_main_stdin_write();
+		schedule_local_sock_write(sock->dest);
 	return G_SOURCE_CONTINUE;
 }
 
-static gboolean terminate_conn_sock(struct conn_sock_s *sock)
+static gboolean terminate_remote_sock(struct remote_sock_s *sock)
 {
-	conn_sock_shutdown(sock, SHUT_RD);
-	if (mainfd_stdin >= 0 && opt_stdin) {
-		if (!opt_leave_stdin_open) {
-			close(mainfd_stdin);
-			mainfd_stdin = -1;
+	remote_sock_shutdown(sock, SHUT_RD);
+	if (SOCK_IS_CONSOLE(sock->sock_type)) {
+		if (sock->dest->readers == NULL || sock->dest->readers->len == 0) {
+			if (*(sock->dest->fd) >= 0 && opt_stdin) {
+				if (!opt_leave_stdin_open) {
+					close(*(sock->dest->fd));
+					*(sock->dest->fd) = -1;
+				} else {
+					ninfo("Not closing input");
+				}
+			}
 		} else {
-			ninfo("Not closing input");
+			ninfo("Not closing input - still have open sockets");
```

Comment:
what's the motivation behind this change? I believe it may be causing crio tests to fail https://storage.googleapis.com/origin-federated-results/pr-logs/pull/cri-o_cri-o/4187/test_pull_request_crio_critest_fedora/19310/artifacts/journals/crio.service

Reply:
I think the idea was to distinguish between the two branches. (line 334 vs line 338). So I could tell during testing which branch it ended up on.

!opt_leave_stdin_open isn't an error condition.
Technically line 338 it's left open because there are still readers.

If you want to change it back or your tests are looking for specific words, obvi it won't break anything.

Reply:
Is your question regarding the verbiage, or that a message is generated at all?

Reply:
applied test commit